Set default bootloader arguments, removing nomodeset and nofb

pdmars opened the pull request:
After updating these I verified by mounting the raw image and inspecting `/mnt/boot/grub/grub.cfg` for the new boot args:
```
linux   /boot/vmlinuz-6.8.0-55-generic root=LABEL=cloudimg-rootfs ro  console=tty0 console=ttyS0,115200 no_timer_check gfxpayload=text pci=realloc=off
```

`nofb` and `nomodeset` have been removed, with `gfxpayload=text pci=realloc=off` set.

The previous config was set with:
```
linux   /boot/vmlinuz-6.8.0-55-generic root=LABEL=cloudimg-rootfs ro  console=tty0 console=ttyS0,115200 no_timer_check nofb nomodeset gfxpayload=text
```

I also checked for the presence of `linux-image-generic`:
```
$ dpkg-query --admindir=/mnt/var/lib/dpkg -W linux-image-generic
linux-image-generic	6.8.0-55.57
```

I only built ubuntu24.04 so far to check this. Not sure if I should be building something else to verify these.

I haven't pushed these changes to glance and booted a node to verify on actual hardware. Should we do that before merging? Or do that once we get the other image changes in (e.g. AMD ROCm updates)?

msherman64 commented:
slight rewording of what I dropped in slack, a few additional checks:
1. I _think_ `modinfo qede` should return something useful
2. it would be good to also check the 20.04 and 22.04 images
3. and finally to check if other images (probably the cuda ones) also have a consistent cmdline set

Additionally, when `nomodeset` is removed from the cuda images, we'll need to check that `nouveau` is still blacklisted, as it will interfere with the cuda drivers if loaded

The end result of that, to me, feels like we'd end up with a set of "current" ubuntu images built and pushed to the new distribution bucket?

msherman64 commented:
on a second reading, to actually answer your question, (what should we test before merging, not "what should we test before publishing new images")

in an ideal world, I'd lean towards having images tested on real hw, and published into the public catalog prior to merging commits here, such that commits on the main branch represent working states. buuuut, we're not there yet in terms of workflow, and the repo already has current, broken image configs in it (mostly centos).

So instead, I might suggest sanity checking 20.04, 22.04, and 24.04 on a `compute_zen3` node at tacc, since that tends to expose e.g. missing network drivers.

but I'd be ok merging before testing cuda images and rocm image, as they both might have changes that also affect kernel cmdline.

And we can figure out a way to reconcile repo state with "what images are known good" down the road, maybe via tagging, a  staging branch, or something else.

(this is all pretty hand-wavy, just my thought on it)

pdmars commented:
Ok, that sounds like a good plan to me. I'll get those images created and boot them on that hardware, then we can merge this, and move on to the rocm changes and testing that stuff.

Once we get all these changes in place, we can move on to generating all the images and pushing them to the new bucket for distribution and expand on our testing/reconciling repo state.

pdmars commented:
@msherman64 Ok it looks like the cuda images were already adding `pci=realloc=off` so I removed the duplicate there since we added it to all Ubuntu images. I also checked that Ubuntu 20, 22, 24 all created the same args (as well as a the cuda image now). I was able to launch ubuntu 20, 22, 24 on `compute_zen3` and verify the args there as well as `modinfo qede`:
```
$ modinfo qede
filename:       /lib/modules/6.8.0-55-generic/kernel/drivers/net/ethernet/qlogic/qede/qede.ko.zst
license:        GPL
description:    QLogic FastLinQ 4xxxx Ethernet Driver
srcversion:     A667968853B63751554F859
alias:          pci:v00001077d00008090sv*sd*bc*sc*i*
alias:          pci:v00001077d00008070sv*sd*bc*sc*i*
alias:          pci:v00001077d00001664sv*sd*bc*sc*i*
alias:          pci:v00001077d00001656sv*sd*bc*sc*i*
alias:          pci:v00001077d00001654sv*sd*bc*sc*i*
alias:          pci:v00001077d00001644sv*sd*bc*sc*i*
alias:          pci:v00001077d00001636sv*sd*bc*sc*i*
alias:          pci:v00001077d00001666sv*sd*bc*sc*i*
alias:          pci:v00001077d00001634sv*sd*bc*sc*i*
depends:        qed
retpoline:      Y
intree:         Y
name:           qede
vermagic:       6.8.0-55-generic SMP preempt mod_unload modversions 
sig_id:         PKCS#7
signer:         Build time autogenerated kernel key
sig_key:        06:43:68:37:39:C4:A1:CF:FE:BE:56:97:CE:F8:75:6B:18:65:14:26
sig_hashalgo:   sha512
signature:      A8:FA:76:3F:F4:44:A1:4A:64:BA:88:7A:23:33:85:E2:AF:F3:3C:E2:
		D8:88:AA:77:C0:88:6E:18:2D:2A:80:B3:6D:8E:31:5E:FC:07:F7:6A:
		21:65:BE:69:27:9A:A4:31:6D:92:F6:C1:95:49:A9:C2:31:59:EF:48:
		9F:78:44:48:0B:35:87:FD:8A:3B:15:99:5B:64:48:6A:B0:28:1D:5B:
		99:13:19:35:5C:A9:26:5B:29:51:A9:50:7C:AE:DD:5A:17:35:6A:C0:
		69:5E:67:E6:D4:12:4A:0D:50:00:7E:50:CE:F3:0D:08:5F:5D:F4:AE:
		BC:66:A3:2A:61:09:A0:51:5A:61:1E:D1:E8:75:54:39:82:1C:DA:7A:
		21:4F:80:BE:9F:24:02:4D:49:55:B2:30:06:21:FC:60:1C:91:FB:01:
		6B:8D:DB:02:32:40:DF:0D:A8:A4:9A:D9:CA:0D:45:32:E8:AD:F5:BB:
		D5:48:81:3A:D0:3D:9D:32:C6:A3:75:66:F1:0E:B3:90:E8:86:28:82:
		F3:DF:78:D1:B0:85:C7:1A:8D:14:27:40:0E:84:96:7E:5F:BE:BF:A2:
		7C:53:BB:40:4E:6A:52:EF:8C:FD:51:F7:28:34:B2:5A:52:34:C3:ED:
		9E:D6:A9:B5:AD:63:68:00:A8:6E:80:20:E3:4D:CF:41:E1:EE:64:A0:
		48:24:B1:16:94:AF:AE:41:6F:68:22:79:BD:DB:5D:AD:18:D4:C3:CD:
		1C:95:EF:46:C7:81:7D:8A:75:02:9C:F0:7C:DC:85:E9:83:B8:22:18:
		DE:65:3C:93:0F:10:36:9F:ED:BD:6A:BF:69:3A:6B:58:F3:2F:4E:14:
		9B:0A:DF:24:12:77:E8:02:F3:C8:06:70:98:F3:5A:0B:E3:9D:07:83:
		52:23:6B:71:CB:98:46:31:84:3C:AE:B8:B1:78:43:4C:4B:9C:07:5F:
		83:2D:FA:38:B1:15:6C:7C:B0:68:D7:43:89:52:8F:C8:D2:28:E1:C4:
		A6:BA:88:12:C6:42:0B:1F:DF:16:3A:8A:05:BE:2D:5C:FD:54:A2:33:
		21:EF:BE:06:47:4D:FF:92:13:C9:17:39:CF:81:50:27:08:49:86:8C:
		4D:CD:53:A7:C2:78:21:54:49:C6:78:50:55:16:C9:1D:9E:13:3B:07:
		22:8B:BC:A2:5B:89:15:0B:E0:BC:5E:26:37:09:1D:C7:02:64:DB:AE:
		EE:77:DE:4F:33:55:C9:8C:B7:5E:11:E2:5B:57:11:14:AF:AA:49:13:
		EF:B8:8A:60:07:65:05:57:19:D8:B6:2B:DB:2E:F5:BE:F0:04:B2:A9:
		74:CC:9C:F5:56:64:11:0A:9F:8B:48:97
parm:           debug: Default debug msglevel (uint)
```

So I think these are good for the first step, then I can move on to AMD ROCm and CUDA changes/testing before building/pushing all images to the new container.

msherman64 commented:
Sounds good!

msherman64 commented on elements/cc-ubuntu-cuda/environment.d/90-linux-cmdline:
```diff
@@ -6,8 +6,3 @@ fi
 set -eu
 set -x
 set -o pipefail
-
-# pci=realloc=off is a fix for A100 nodes which fail to load Nvidia drivers
-# if 4 or more GPUs are installed
-DIB_BOOTLOADER_DEFAULT_CMDLINE="${DIB_BOOTLOADER_DEFAULT_CMDLINE} pci=realloc=off"
-
```
we should just delete this file entirely if it's going to be empty

msherman64 reviewed (approved):
minor cleanup on the cuda image, otherwise LGTM
